plugin-server/src/worker/vm/utils.ts

```diff
@@ -72,3 +73,66 @@ export const addPublicJobIfNotExists = async (
         'addPublicJobIfNotExists'
     )
 }
+
+type BufferOptions = {
+    limit: number
+    timeoutSeconds: number
+    onFlush: (objects: any[], points: number) => void | Promise<void>
+}
+
+export function createBuffer(opts: Partial<BufferOptions>, statsd?: StatsD) {
```

Comment:
this is essentially a copy 🍝 from https://github.com/PostHog/plugin-contrib/blob/main/src/buffer.ts with instrumentation added

Reply:
checks out